clang/lib/AST/ExprConstant.cpp

```diff
@@ -16268,6 +16353,25 @@ class VoidExprEvaluator
     }
   }
 
+  bool VisitAtomicExpr(const AtomicExpr *E) {
+    if (!EvaluateAtomicOrder(E, Info))
+      return false;
+
+    switch (E->getOp()) {
+    default:
+      return Error(E);
```

Comment:
Does this do the correct thing for code like:
```
constexpr int foo(const int *ptr) {
  (void)__atomic_load_n(ptr, __ATOMIC_RELAXED);
  return 12;
}
constexpr int i = 12;
static_assert(foo(&i) == 12);
```
I would expect that to compile and pass the static assertion.